packages/veritone-widgets/src/redux/modules/mediaDetails/index.js

```diff
@@ -0,0 +1,657 @@
+        success: !(warn || error) || null,
+        error: error ? errorMessage : null,
+        warning: warn || null,
+        tdo: tdo
+      }
+    };
+  },
+  [UPDATE_TDO](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null
+      }
+    };
+  },
+  [UPDATE_TDO_COMPLETE](state, { payload, meta: { warn, error, widgetId } }) {
+    const errorMessage = get(error, 'message', error);
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: !(warn || error) || null,
+        error: error ? errorMessage : null,
+        warning: warn || null,
+        tdo: payload
+      }
+    };
+  },
+  [LOAD_TDO_CONTENT_TEMPLATES_COMPLETE](
+    state,
+    { payload, meta: { warn, error, widgetId } }
+  ) {
+    const errorMessage = get(error, 'message', error);
+    const tdoContentTemplates = {};
+    if (payload && payload.records) {
+      payload.records.forEach(asset => {
+        if (!asset.sourceData || !asset.sourceData.schema) {
+          return;
+        }
+        const contentTemplate = Object.assign({}, asset.sourceData.schema);
+        if (asset.transform) {
+          contentTemplate.data = JSON.parse(asset.transform);
+        }
+        if (contentTemplate.dataRegistry && contentTemplate.dataRegistry.name) {
+          contentTemplate.name = contentTemplate.dataRegistry.name;
+          delete contentTemplate.dataRegistry;
+        }
+        // keep asset id on the content template for asset CRUD
+        contentTemplate.assetId = asset.id;
+        tdoContentTemplates[contentTemplate.id] = contentTemplate;
+      });
+    }
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: !(warn || error) || null,
+        error: error ? errorMessage : null,
+        warning: warn || null,
+        tdoContentTemplates: tdoContentTemplates
+      }
+    };
+  },
+  [UPDATE_TDO](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null
+      }
+    };
+  },
+  [UPDATE_TDO_CONTENT_TEMPLATES](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null
+      }
+    };
+  },
+  [UPDATE_TDO_CONTENT_TEMPLATES_FAILURE](state, { error, meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        updateTdoContentTemplatesError: error
+      }
+    };
+  },
+  [LOAD_CONTENT_TEMPLATES](state, { meta: { widgetId } }) {
+    return {
+      ...state,
+      [widgetId]: {
+        ...state[widgetId],
+        success: null,
+        error: null,
+        warning: null
+      }
+    };
+  },
+  [LOAD_CONTENT_TEMPLATES_COMPLETE](
```

Comment:
see my comment for `LOAD_ENGINE_CATEGORIES_COMPLETE`

Reply:
done